skimage/segmentation/_clear_border.py

```diff
@@ -58,49 +67,54 @@ def clear_border(labels, buffer_size=0, bgval=0, in_place=False, mask=None):
            [0, 0, 0, 0, 0, 0, 0, 0, 0]])
 
     """
-    image = labels
-
-    if any((buffer_size >= s for s in image.shape)) and mask is None:
+    if any((buffer_size >= s for s in labels.shape)) and mask is None:
         # ignore buffer_size if mask
-        raise ValueError("buffer size may not be greater than image size")
+        raise ValueError("buffer size may not be greater than labels size")
+
+    if out is not None:
+        np.copyto(out, labels, casting='no')
+        in_place = True
+
+    if not in_place:
+        out = labels.copy()
+    elif out is None:
+        out = labels
 
     if mask is not None:
-        err_msg = "image and mask should have the same shape but are {} and {}"
-        assert image.shape == mask.shape, \
-               err_msg.format(image.shape, mask.shape)
+        err_msg = ("labels and mask should have the same shape but "
+                   "are {} and {}")
+        if out.shape != mask.shape:
+            raise(ValueError, err_msg.format(out.shape, mask.shape))
         if mask.dtype != bool:
             raise TypeError("mask should be of type bool.")
         borders = ~mask
     else:
         # create borders with buffer_size
-        borders = np.zeros_like(image, dtype=bool)
+        borders = np.zeros_like(out, dtype=bool)
         ext = buffer_size + 1
         slstart = slice(ext)
         slend = slice(-ext, None)
-        slices = [slice(s) for s in image.shape]
-        for d in range(image.ndim):
-            slicedim = list(slices)
-            slicedim[d] = slstart
-            borders[tuple(slicedim)] = True
-            slicedim[d] = slend
-            borders[tuple(slicedim)] = True
-    # Re-label, in case we are dealing with a binary image
+        slices = [slice(None) for _ in out.shape]
+        for d in range(out.ndim):
+            slices[d] = slstart
+            borders[tuple(slices)] = True
+            slices[d] = slend
+            borders[tuple(slices)] = True
+            slices[d] = slice(None)
+
+    # Re-label, in case we are dealing with a binary out
     # and to get consistent labeling
-    labels = label(image, background=0)
-    number = np.max(labels) + 1
+    labels, number = label(out, background=0, return_num=True)
 
     # determine all objects that are connected to borders
     borders_indices = np.unique(labels[borders])
     indices = np.arange(number + 1)
     # mask all label indices that are connected to borders
     label_mask = np.in1d(indices, borders_indices)
     # create mask for pixels to clear
-    mask = label_mask[labels.ravel()].reshape(labels.shape)
-
-    if not in_place:
-        image = image.copy()
+    mask = label_mask[labels.reshape(-1)].reshape(labels.shape)
```

Comment:
Is there a benefit to this `reshape(-1)` form vs. ravel? I think neither makes a copy, right? (IIRC `flatten` copies, but `ravel` does not)

Reply:
That's just what [`ravel`](https://numpy.org/doc/stable/reference/generated/numpy.ravel.html) doc says in the note :slightly_smiling_face:. It is really not important, but well, it became one of my habits... I can revert it if you wish.

Reply:
I don't have a strong preference either way.